src/data_structure/Board.java

```diff
@@ -275,43 +235,49 @@ private void setPossibilities() {
 
             possibilities = PossiblesSumCombinations.getPossibilities(column.getSum(), column.getFieldCount());
             for (FieldWritable j : column.getFields()) {
-                if (j.getState() != FieldWritable.State.FILLED)
+                if (j.getState() == FieldWritable.State.UNFILLED)
                     j.setPossibilities(possibilities);
             }
         }
 
         for (FieldInfo row : rowsInfo) {
             possibilities = PossiblesSumCombinations.getPossibilities(row.getSum(), row.getFieldCount());
             for (FieldWritable writable : row.getFields()) {
-                if (writable.getState() != FieldWritable.State.FILLED) {
+                if (writable.getState() == FieldWritable.State.UNFILLED) {
                     possibilitiesRow = writable.getPossibilities();
                     for (int k = 0; k < 9; ++k)
                         crossPossibilities[k] = possibilities[k] & possibilitiesRow[k];
                     writable.setPossibilities(crossPossibilities);
                 }
             }
         }
+    }
 
-//      this should consider filled fields influence
-//      and should not fill fields
+    private void setPossibilitiesBasedOnFilledField(){
         for (FieldInfo column : columnsInfo) {
             for (FieldWritable writable : column.getFields()) {
-                if (writable.getState() == FieldWritable.State.UNFILLED) {
-                    updatePossibilitiesByFilledFields(column.getX(), writable.getRowFieldInfo().getY());
+                if (writable.getState() == FieldWritable.State.FILLED) {
+                    updatePossibilitiesByFilledFields(writable);
                 }
             }
         }
     }
 
-    private void updatePossibilitiesByFilledFields(int x, int y) {
-        for (FieldWritable fieldInColumn : gameBoard[x][y].getWritable().getColumnFieldInfo().getFields()) {
-            if (fieldInColumn != gameBoard[x][y].getWritable() && fieldInColumn.getState() == FieldWritable.State.FILLED) {
-                gameBoard[x][y].getWritable().revokePossibility(fieldInColumn.getValue());
-            }
+    private void updatePossibilitiesByFilledFields(FieldWritable writable) {
+        for (FieldWritable fieldInColumn : writable.getColumnFieldInfo().getFields()) {
+            updatePossibilitiesInField(writable, fieldInColumn);
         }
-        for (FieldWritable fieldInRow : gameBoard[x][y].getWritable().getRowFieldInfo().getFields()) {
-            if (fieldInRow != gameBoard[x][y].getWritable() && fieldInRow.getState() == FieldWritable.State.FILLED) {
-                gameBoard[x][y].getWritable().revokePossibility(fieldInRow.getValue());
+        for (FieldWritable fieldInRow : writable.getRowFieldInfo().getFields()) {
+            updatePossibilitiesInField(writable, fieldInRow);
+        }
+    }
+
+    private void updatePossibilitiesInField(FieldWritable writable, FieldWritable fieldInRow) {
```

Comment:
Here `fieldInRow`, on L268 `fieldInColumn`, in fact, what is it? Mayby name should be more generic